
message man_message {
  required string message_type = 1;
  
  message FullMANDB {
    message TableType_node {
      optional int32 id = 1;
      optional string ip = 2;
    }
    repeated TableType_node node = 1;
    
    message TableType_controller {
      optional int32 id = 1;
      optional int32 address = 2;
      optional int32 node_id = 3;
    }
    repeated TableType_controller controller = 2;
    
    message TableType_point {
      optional int32 id = 1;
      optional string type = 2;
      optional int32 controller_id = 3;
      optional string reader = 4;
    }
    repeated TableType_point point = 3;
    
    message TableType_user_group {
      optional int32 id = 1;
      optional string name = 2;
    }
    repeated TableType_user_group user_group = 4;
    
    message TableType_user {
      optional int32 id = 1;
      optional string name = 2;
      optional string surname = 3;
      optional string patronymic = 4;
      optional string gender = 5;
    }
    repeated TableType_user user = 5;
    
    message TableType_em_marine_card {
      optional int32 id = 1;
      optional int64 code = 2;
      optional int32 user_id = 3;
    }
    repeated TableType_em_marine_card em_marine_card = 6;
    
    message TableType_point_group {
      optional int32 id = 1;
      optional string name = 2;
    }
    repeated TableType_point_group point_group = 7;
    
    message TableType_script {
      optional int32 id = 1;
      optional string condition_code = 2;
    }
    repeated TableType_script script = 8;
    
    message TableType_access {
      optional int32 id = 1;
      optional int32 script_id = 2;
      optional int32 point_group_id = 3;
      optional int32 user_group_id = 4;
    }
    repeated TableType_access access = 9;
    
    message TableType_point_group_link {
      optional int32 id = 1;
      optional int32 point_id = 2;
      optional int32 point_group_id = 3;
    }
    repeated TableType_point_group_link point_group_link = 10;
    
    message TableType_user_group_link {
      optional int32 id = 1;
      optional int32 user_id = 2;
      optional int32 user_group_id = 3;
    }
    repeated TableType_user_group_link user_group_link = 11;
    
    message TableType_log {
      optional int32 id = 1;
      optional string time = 2;
      optional string decision = 3;
      optional int32 point_id = 4;
      optional int64 code = 5;
      optional string code_type = 6;
    }
    repeated TableType_log log = 12;
    
    message TableType_database_send_log {
      optional int32 id = 1;
      optional string begin_time = 2;
      optional string end_time = 3;
      optional int32 node_id = 4;
    }
    repeated TableType_database_send_log database_send_log = 13;
    
  }
  optional FullMANDB db = 2;
}
